src/Rule/AllowComparingOnlyComparableTypesRule.php

```diff
@@ -79,7 +80,23 @@ private function isComparable(Type $type): bool
         $stringType = new StringType();
         $dateTimeType = new ObjectType(DateTimeInterface::class);
 
-        return $this->containsOnlyTypes($type, [$intType, $floatType, $stringType, $dateTimeType]);
+        if ($this->containsOnlyTypes($type, [$intType, $floatType, $stringType, $dateTimeType])) {
+            return true;
+        }
+
+        if (!$type->isConstantArray()->yes() || !$type->isList()->yes()) {
```

Comment:
What is the list condition doing here? When I remove it or change this to &&, no test fails.